### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by faisalakb**
In this project apply the following features : :100: 

Create a method called current_user in your ApplicationController to make current user data available to all controllers. :+1: 

Create forms to perform the following functions: :+1: 

    Creates a Post on behalf of the current_user
    Create a comment on behalf of the current_user

Allow Users to add likes to Posts. :+1:

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  def create
+    set_current_user
+    @post = @current_user.posts.build(post_params)
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+
+    if @post.save
+      redirect_to user_post_path(@current_user.id, @post.id), notice: 'Post was successfully Created'
+    else
```
Comment (Munsa1) [inline app/controllers/posts_controller.rb:25]: - [ ] There is an error preventing the user from creating a POST and therefore I am unable to check the rest os the apps functionality. Kindly fix this/

ScreenCast 
[cinnamon-2024-01-27T003255+0200.webm](https://github.com/faisalakb/R_blog_app/assets/81932478/e8bd28dc-19fe-4dc6-9b34-0544ea129d08)

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @faisalakb 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Added Forms :heavy_check_mark: 
Correct Git Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by faisalakb** [comment]
Please check i  address the above comments

**Review by prolajumokeoni — APPROVED** [review]
Hi @faisalakb 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - Updated create new post method ✅ 
   

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**